test_scripts/Polices/External_UCS/006_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_on_struct.lua

```diff
@@ -0,0 +1,148 @@
+--
+-- Preconditions:
+-- 1. Stop SDL
+-- 2. Modify PreloadedPolicyTable (add 'disallowed_by_external_consent_entities_on' section)
+--
+-- Steps:
+-- 1. Start SDL
+-- 2. Check SDL status
+-- 3. Register app
+-- 4. Activate app
+-- 5. Verify PTSnapshot
+--
+-- Expected result:
+-- a. Status = 1 (SDL is running)
+-- b. PTSnapshot contains 'disallowed_by_external_consent_entities_on' section
+--
+-- Note: Script is designed for EXTERNAL_PROPRIETARY flow
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+config.defaultProtocolVersion = 2
+
+--[[ Required Shared Libraries ]]
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local commonPreconditions = require('user_modules/shared_testcases/commonPreconditions')
+local sdl = require('SDL')
+local testCasesForExternalUCS = require('user_modules/shared_testcases/testCasesForExternalUCS')
+
+--[[ Local variables ]]
+local grpId = "Location-1"
+local checkedSection = "disallowed_by_external_consent_entities_on"
+
+--[[ General Precondition before ATF start ]]
+commonFunctions:SDLForceStop()
+commonSteps:DeleteLogsFileAndPolicyTable()
+commonPreconditions:BackupFile("sdl_preloaded_pt.json")
+testCasesForExternalUCS.removePTS()
+
+--[[ General Settings for configuration ]]
+Test = require("user_modules/connecttest_resumption")
+require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:CheckSDLStatus_1_RUNNING()
+  testCasesForExternalUCS.checkSDLStatus(self, sdl.RUNNING)
+end
+
+function Test:StopSDL()
+  testCasesForExternalUCS.ignitionOff(self)
+end
+
+function Test:CheckSDLStatus_2_STOPPED()
+  testCasesForExternalUCS.checkSDLStatus(self, sdl.STOPPED)
+end
+
+function Test.RemoveLPT()
+  testCasesForExternalUCS.removeLPT()
+end
+
+function Test.UpdatePreloadedPT()
+  local updateFunc = function(preloadedTable)
+    preloadedTable.policy_table.functional_groupings[grpId][checkedSection] = {
+      {
+        entityID = 128,
+        entityType = 0
+      }
+    }
+  end
+  testCasesForExternalUCS.updatePreloadedPT(updateFunc)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+function Test.StartSDL()
+  StartSDL(config.pathToSDL, config.ExitOnCrash)
+  os.execute("sleep 5")
+end
+
+function Test:CheckSDLStatus_3_RUNNING()
+  testCasesForExternalUCS.checkSDLStatus(self, sdl.RUNNING)
+end
+
+function Test:InitHMI()
+  self:initHMI()
+end
+
+function Test:InitHMI_onReady()
+  self:initHMI_onReady()
+end
+
+function Test:ConnectMobile()
+  self:connectMobile()
+end
+
+function Test:StartSession()
+  testCasesForExternalUCS.startSession(self, 1)
+end
+
+function Test:RAI()
+  testCasesForExternalUCS.registerApp(self, 1)
```

Comment:
@istoimenova, Not accepted.
Due to, req-ts: 
a) "[RegisterAppInterface] RegisterAppInterface and HMILevel"
b) "[RegisterAppInterface] Order of request/response/notifications on registering an application"

Currently there is an issue created against a): "OnHMIStatus is sent before response to RegisterAppInterface request"